### ISSUE TITLE
Update to capacitor 3 or 4

### ISSUE THREAD

**Issue by zven** [sitcomlab/simport-learning-app#268]
- [ ] Direct jump from capacitor 2 to capacitor 4 possible?
- [ ] quick test with capacitor 4: What is new? What breaks? What todos emerge?
- [ ] If too much work: capacitor 3 update more manageable?

Note: This card will most likely need to be split eventually.

**Comment by felixerdy** [comment]
I just checked what we have to to in order to update to a new capacitorjs version.

__Migrating to Capacitor 4__
After reading the [blog post of the Capacitor 4 announcement](https://ionic.io/blog/announcing-capacitor-4-0) I just followed their instructions to migrate our App to capacitor 4 using the CLI as well as the VSCode extension (which shows additional recommendations for multiple packages. Quite nice!)
The migration script itself ran trough and it seems like upgrading isn't too hard

__Packages__
Following the recommendations from the [VSCode Extension](https://marketplace.visualstudio.com/items?itemName=ionic.ionic) there are some packages that should either be updated or exchanged (to replace cordova packages with capacitor ones).
I updated and replaced some packages (especially the `@ionic-native/` ones which are cordova packages).
 
<img width="1232" alt="Screenshot 2023-01-16 at 16 31 43" src="https://user-images.githubusercontent.com/12086004/212714696-97a2c610-00dc-4ce5-b755-e74d278d8f7a.png">

Updating the packages led to code changes as well. While some packages could be exchanged as is, others needed a little bit of more work. SQLite queries are now wrapped in `try .. catch` blocks

<img width="1194" alt="Screenshot 2023-01-16 at 16 35 31" src="https://user-images.githubusercontent.com/12086004/212715456-573be31a-f6af-42e6-b08a-2f741f54404e.png">

Overall, it's not too much work to migrate the packages to the new ones. Some even seem to simplify code parts (e.g. automatically requesting permissions on android)

__TODO__
There is still some work to do. I migrated in a quick and dirty way, not focusing on code style etc. Furthermore, as some packages were replaced, we need to check if everything is still working.
Additionally, we need to decide which Android / iOS Versions we want to support and check what the minimum versions for capacitor 4 are.

__Conclusion__
I think we can migrate our application directly to Capacitor 4. The majority of the work would be updating to the latest Plugins.
